src/mining_minigame.c

```diff
@@ -0,0 +1,3395 @@
+                return;
+            }
+            gMain.state++;
+            break;
+
+        case STATE_LOAD_BGS:
+            if (Mining_LoadBgGraphics() == TRUE)
+            {
+                InitMiningWindows();
+                gMain.state++;
+            }
+            break;
+
+        case STATE_LOAD_SPRITES:
+            if (!gPaletteFade.active)
+            {
+                InitBuriedItems();
+                Mining_LoadSpriteGraphics();
+                gMain.state++;
+            }
+            break;
+
+        case STATE_WAIT_FADE:
+            CreateTask(Task_MiningWaitFadeIn, 0);
+            gMain.state++;
+            break;
+
+        case STATE_FADE:
+            BeginNormalPaletteFade(PALETTES_ALL, 0, 16, 0, RGB_BLACK);
+            gMain.state++;
+            break;
+
+        case STATE_SET_CALLBACKS:
+            SetVBlankCallback(Mining_VBlankCB);
+            SetMainCallback2(Mining_MainCB);
+            break;
+    }
+}
+
+static bool8 Mining_InitBgs(void)
+{
+    const u32 TILEMAP_BUFFER_SIZE = (1024 * 2);
+
+    ResetAllBgsCoordinates();
+
+    sMiningUiState->sBg1TilemapBuffer = AllocZeroed(TILEMAP_BUFFER_SIZE);
+    sMiningUiState->sBg2TilemapBuffer = AllocZeroed(TILEMAP_BUFFER_SIZE);
+    sMiningUiState->sBg3TilemapBuffer = AllocZeroed(TILEMAP_BUFFER_SIZE);
+
+    if (sMiningUiState->sBg3TilemapBuffer == NULL)
+        return FALSE;
+    else if (sMiningUiState->sBg2TilemapBuffer == NULL)
+        return FALSE;
+    else if (sMiningUiState->sBg1TilemapBuffer == NULL)
+        return FALSE;
+
+    ResetBgsAndClearDma3BusyFlags(0);
+
+    InitBgsFromTemplates(0, sMiningBgTemplates, NELEMS(sMiningBgTemplates));
+
+    SetBgTilemapBuffer(1, sMiningUiState->sBg1TilemapBuffer);
+    SetBgTilemapBuffer(2, sMiningUiState->sBg2TilemapBuffer);
+    SetBgTilemapBuffer(3, sMiningUiState->sBg3TilemapBuffer);
+
+    ScheduleBgCopyTilemapToVram(1);
+    ScheduleBgCopyTilemapToVram(2);
+    ScheduleBgCopyTilemapToVram(3);
+
+    ShowBg(0);
+    ShowBg(2);
+    ShowBg(3);
+
+    return TRUE;
+}
+
+static void Task_Mining_WaitFadeAndBail(u8 taskId)
+{
+    if (!gPaletteFade.active)
+    {
+        SetMainCallback2(sMiningUiState->leavingCallback);
+        Mining_FreeResources();
+        DestroyTask(taskId);
+    }
+}
+
+static void Mining_MainCB(void)
+{
+    RunTasks();
+    AnimateSprites();
+    BuildOamBuffer();
+    DoScheduledBgTilemapCopiesToVram();
+}
+
+static void MoveItemSprites(s16 dx, s16 dy)
+{
+    u32 i;
+
+    if (sMiningUiState->toggleShakeDuringAnimation == FALSE)
+    {
+        for (i=0;i<MAX_SPRITES;i++)
+        {
+            gSprites[i].x += dx;
+            gSprites[i].y += dy;
+        }
+    }
+}
+
+static void MiningUi_Shake(u8 taskId)
+{
+    switch(sMiningUiState->shakeState)
+    {
+        case 0: // Left 1 - Down 1
+            MakeCursorInvisible();
+            if (!IsStressLevelMax() && Random() % 100 < 20) // 20 % chance of not shaking the screen
```

Comment:
```suggestion
            if (!IsStressLevelMax() && RandomPercentage(RNG_MINING_SHAKE, 20)) // 20 % chance of not shaking the screen
```
Use a tagged call to `RandomPercentage` instead, tag needs to be added to `include/random.h`.